contracts/JoeMakerV2.sol

```diff
@@ -0,0 +1,231 @@
+        uint256 amountJOE
+    );
+
+    /* ========== CONSTRUCTOR ========== */
+
+    constructor(
+        address _factory,
+        address _bar,
+        address _joe,
+        address _wavax
+    ) public {
+        factory = IJoeFactory(_factory);
+        bar = _bar;
+        joe = _joe;
+        wavax = _wavax;
+    }
+
+    /* ========== External Functions ========== */
+
+    // F3: _convert is separate to save gas by only checking the 'onlyEOA' modifier once in case of convertMultiple
+    // F6: There is an exploit to add lots of JOE to the bar, run convert, then remove the JOE again.
+    //     As the size of the JoeBar has grown, this requires large amounts of funds and isn't super profitable anymore
+    //     The onlyEOA modifier prevents this being done with a flash loan.
+    function convert(address token0, address token1) external onlyEOA {
+        _convert(token0, token1);
+    }
+
+    function convertMultiple(address[] calldata token0, address[] calldata token1) external onlyEOA {
+        // TODO: This can be optimized a fair bit, but this is safer and simpler for now
+        uint256 len = token0.length;
+        for (uint256 i = 0; i < len; i++) {
+            _convert(token0[i], token1[i]);
+        }
+    }
+
+    /* ========== Modifiers ========== */
+
+    // It's not a fool proof solution, but it prevents flash loans, so here it's ok to use tx.origin
+    modifier onlyEOA() {
+        // Try to make flash-loan exploit harder to do by only allowing externally owned addresses.
+        require(msg.sender == tx.origin, "JoeMakerV2: must use EOA");
+        _;
+    }
+
+    /* ========== Public Functions ========== */
+
+    function bridgeFor(address token) public view returns (address bridge) {
+        bridge = _bridges[token];
+        if (bridge == address(0)) {
+            bridge = wavax;
+        }
+    }
+
+    /* ========== Internal Functions ========== */
+
+    function _convert(address token0, address token1) internal {
+        // Interactions
+        // S1 - S4: OK
+        IJoePair pair = IJoePair(factory.getPair(token0, token1));
+        require(address(pair) != address(0), "JoeMakerV2: Invalid pair");
+        // balanceOf: S1 - S4: OK
+        // transfer: X1 - X5: OK
+        IERC20(address(pair)).safeTransfer(address(pair), pair.balanceOf(address(this)));
+
+        // X1 - X5: OK
+        // We don't take amount0 and amount1 from here, as it won't take into account reflect tokens.
+        pair.burn(address(this));
+
+        // We get the amount0 and amount1 by their respective balance of JoeMakerV2.
+        uint256 amount0 = IERC20(token0).balanceOf(address(this));
+        uint256 amount1 = IERC20(token1).balanceOf(address(this));
+
+        // we invert token0 and token1 as well as we'll need them to get their respective balance.
+        if (token0 != pair.token0()) {
+            (amount0, amount1) = (amount1, amount0);
+            (token0, token1) = (token1, token0);
```

Comment:
What's the purpose of this line?

Reply:
It's useless, I didn't see that when I changed the way to retrieve the amounts, there was no need to exchange the tokens or the amounts.